js/page-specific/baltcogo-app/providers/data-provider.js

```diff
@@ -0,0 +1,28 @@
+// http://www.learn-angular.org/#!/lessons/the-provider-recipe
+(function (app, querystringer) {
+  'use strict';
+
+  function Data(arr) {
+    this.endpoints = {};
+
+    for (var i = 0; i < arr.length; i++) {
+      var obj = arr[i];
+      var key = Object.keys(obj)[0];
+
+      this.endpoints[key] = obj[key];
+    }
+  }
+
+  app.provider('data', function () {
+    var urlArr = null;
+
+    this.setUrls = function (keyValArr) {
```

Comment:
Elsewhere in the code, we follow the **var foo = function () { ... }; return { foo: foo };** pattern. I see the original author likes using _this_ all over the place, but I think should should still be adapted into our style.